java-checks/src/main/java/org/sonar/java/checks/JavaFootprint.java

```diff
@@ -30,7 +30,9 @@ public final class JavaFootprint implements LanguageFootprint {
   private final Set<Detector> detectors = new HashSet<>();
 
   public JavaFootprint() {
-    detectors.add(new EndWithDetector(0.95, '}', ';', '{'));
+    // Reduce probability for semicolon to avoid false positives with natural language
+    detectors.add(new EndWithDetector(0.95, '}', '{'));
+    detectors.add(new EndWithDetector(0.3, ';'));
```

Comment:
I strongly disagree with the change of probability from `0.95` to `0.3` for a `;` at the end of the line when detecting Java code! Java is not Kotlin, the lines almost always end with a `;`